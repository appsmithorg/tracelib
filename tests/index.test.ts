import Tracelib from '../src/index'
import JANK_TRACE_LOG from './__fixtures__/jankTraceLog.json'

test('should contain traceLog', () => {
    const sampleTrace = new Tracelib({ foo: 'bar' })
    expect(sampleTrace.tracelog).toEqual({ foo: 'bar' })
})

test('should get FPS', () => {
    const trace = new Tracelib(JANK_TRACE_LOG)
    const result = trace.getFPS()
    expect(result).toMatchSnapshot()
})

test('should get summary data', () => {
    const trace = new Tracelib(JANK_TRACE_LOG)
    const result = trace.getSummary()
    expect(result).toMatchSnapshot()
})

test('should get summary data between passed range', () => {
    const trace = new Tracelib(JANK_TRACE_LOG)
    const result = trace.getSummary(289960055.634, 289960729.717)
    expect(result).toMatchSnapshot()
})

test('should get warning counts', () => {
    const trace = new Tracelib(JANK_TRACE_LOG)
    const result = trace.getWarningCounts()
    expect(result).toMatchSnapshot()
})

test('getWarningCounts throws error if mainTrack is missing', () => {
    const trace = new Tracelib([])
    expect(() => trace.getMainThreadEventsLength())
        .toThrow(new Error('MainTrack is missing in traceLog'))
})

test('should get number of events in mainThread', () => {
    const trace = new Tracelib(JANK_TRACE_LOG)
    const result = trace.getMainThreadEventsLength()
    expect(result).toEqual(56244)
})

test('getMainThreadEventsLength throws error if mainTrack is missing', () => {
    const trace = new Tracelib([])
    expect(() => trace.getMainThreadEventsLength())
        .toThrow(new Error('MainTrack is missing in traceLog'))
})
