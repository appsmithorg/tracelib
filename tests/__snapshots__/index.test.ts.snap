// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get FPS 1`] = `
Array [
  182.2821727559685,
  10.307790628308753,
  11.092131244032895,
  11.014792866762287,
  11.00897231503525,
  10.794939328106791,
  10.929081197725838,
  10.815838712204958,
  10.556652274643293,
  10.47987340271033,
  10.926095888726774,
  10.486577179634944,
  10.897876006628481,
  10.839990888916617,
]
`;

exports[`should get summary data 1`] = `
Object {
  "endTime": 289961229.717,
  "idle": 52.38300037384033,
  "other": 9.896000564098358,
  "painting": 69.94999980926514,
  "rendering": 847.373997092247,
  "scripting": 394.4800021648407,
  "startTime": 289959855.634,
}
`;

exports[`should get summary data between passed range 1`] = `
Object {
  "endTime": 289960729.717,
  "idle": 0.6339998841285706,
  "other": 4.653000295162201,
  "painting": 34.8999999165535,
  "rendering": 425.89399832487106,
  "scripting": 208.0020015835762,
  "startTime": 289960055.634,
}
`;

exports[`should get warning counts 1`] = `
Object {
  "ForcedLayout": 4683,
  "ForcedStyle": 4684,
  "LongRecurringHandler": 13,
}
`;
